src/commands/tree_of.rs

```diff
@@ -88,10 +90,24 @@ pub async fn tree_of(matches: &ArgMatches, repo: Repository, config: &Configurat
                             }
                         )
                     },
-                    &|_, node| format!("label = \"{}\" ", node.1.display_name_version()),
+                    &|_, node| format!("label = \"{}\" ", node.1.clone().display_name_version()),
                 );
 
                 println!("{:?}", dot);
+                Ok(())
+            } else if serial_buildorder {
+                let petgraph: DiGraph<Package, DependencyType> = (*dag.dag()).clone().into();
+
+                let topo_sorted = petgraph::algo::toposort(&petgraph, None)
+                    .map_err(|_| Error::msg("Cyclic dependency found!"))?;
+
+                for i in (0..topo_sorted.len()).rev() {
+                    let node = topo_sorted[i];
+                    let package = petgraph.node_weight(node).unwrap();
+                    println!("{}", package.clone().display_name_version());
+                }
```

Comment:
This shouldn't be such a for loop (only makes it more error prone).
`for node in topo_sorted.iter().rev()` should be much better (and I just learned that a for loop is apparently even more idiomatic than `for_each` in such cases (https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.for_each)).